### PR TITLE
[config] Add values for cppstds for conan v2 pipeline

### PR DESCRIPTION
**Pull request by danimtb**
Added list of `cppstd` values for the different compilers.

Currently the support for cpp 23 is not very much supported on any compiler.

### CODE REVIEW

**Review by jcar87 — APPROVED** [review]
LGTM!

File: .c3i/config_v2.yml [hunk 1]
```diff
@@ -103,11 +103,11 @@ configurations:
 
 cppstd:
   apple-clang:
-    "13": ["gnu17"]
+    "13": ["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17", "20", "gnu20"]
```
Comment (prince-chrismc) [inline .c3i/config_v2.yml:106]: Do we really need to build in cpp 98?

File: .c3i/config_v2.yml [hunk 2]
```diff
@@ -103,11 +103,11 @@ configurations:
...
   gcc:
-    "11": ["gnu17"]
+    "11": ["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17", "20", "gnu20"]
```
Comment (danimtb) [inline .c3i/config_v2.yml:108]:
```suggestion
    "11": ["17", "gnu17", "20", "gnu20"]
```

**Review by jcar87 — APPROVED** [review]
Thanks!!!

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**